File: autoscale/src/main/java/com/sequenceiq/periscope/PeriscopeApplication.java
package com.sequenceiq.periscope;

import static com.sequenceiq.periscope.VersionedApplication.versionedApplication;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.actuate.autoconfigure.observation.web.servlet.WebMvcObservationAutoConfiguration;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.autoconfigure.web.servlet.error.ErrorMvcAutoConfiguration;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.EnableAspectJAutoProxy;

import com.sequenceiq.cloudbreak.util.OpenSSLLoaderUtil;

@Configuration
@EnableAspectJAutoProxy(proxyTargetClass = true)
@SpringBootApplication(scanBasePackages = "com.sequenceiq", exclude = { ErrorMvcAutoConfiguration.class, WebMvcObservationAutoConfiguration.class })
public class PeriscopeApplication {

    public static void main(String[] args) {
        OpenSSLLoaderUtil.registerOpenSSLJniProvider();
        if (!versionedApplication().showVersionInfo(args)) {
            if (args.length == 0) {
                SpringApplication.run(PeriscopeApplication.class);
            } else {
                SpringApplication.run(PeriscopeApplication.class, args);
            }
        }
    }

}

File: autoscale/src/main/java/com/sequenceiq/periscope/PeriscopeApplication.kt
package com.sequenceiq.periscope

import com.sequenceiq.periscope.VersionedApplication.versionedApplication

import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.EnableAutoConfiguration
import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Configuration

@Configuration
@EnableAutoConfiguration
@ComponentScan(basePackages = "com.sequenceiq.periscope")
object PeriscopeApplication {

    @JvmStatic fun main(args: Array<String>) {
        if (!versionedApplication().showVersionInfo(args)) {
            if (args.size == 0) {
                SpringApplication.run(PeriscopeApplication::class.java)
            } else {
                SpringApplication.run(PeriscopeApplication::class.java, *args)
            }
        }
    }

}
